build_tools/pytype/check_diff.sh

```diff
@@ -0,0 +1,84 @@
+#!/bin/bash
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Uses git diff to run pytype on changed files.
+
+DIFF_TARGET="${1:-main}"
+echo "Running pycheck against '${DIFF_TARGET?}'"
+
+if [[ "${DIFF_TARGET?}" = "all" ]]; then
+  FILES=$(find -name "*\.py" -not -path "./third_party/*")
+else
+  FILES=$(git diff --name-only "${DIFF_TARGET?}" | grep '.*\.py')
+fi
+
+BASE=$(echo "${FILES?}" | grep -v '^[\./]*integrations/*')
+IREE_TF=$(echo "${FILES?}" | \
+          grep '^[\./]*integrations/tensorflow/bindings/python/pyiree/tf/.*')
+IREE_XLA=$(echo "${FILES?}" | \
+           grep '^[\./]*integrations/tensorflow/bindings/python/pyiree/xla/.*')
+COMPILER=$(echo "${FILES?}" | \
+           grep '^[\./]*integrations/tensorflow/compiler/.*')
+E2E=$(echo "${FILES?}" | grep '^[\./]*integrations/tensorflow/e2e/.*')
```

Comment:
I don't think the `*` in these patterns are doing what you think? These are regexes. They're also only partial regexes (unless you anchor them with the end of line `$`). You don't want to be matching `.//./...././././integrations//////////`. I think this should be

`'^\./integrations/'`


```suggestion
BASE=$(echo "${FILES?}" | grep -v '^\./integrations/')
IREE_TF=$(echo "${FILES?}" | \
          grep '^\./integrations/tensorflow/bindings/python/pyiree/tf/')
IREE_XLA=$(echo "${FILES?}" | \
           grep '^\./integrations/tensorflow/bindings/python/pyiree/xla/')
COMPILER=$(echo "${FILES?}" | \
           grep '^\./integrations/tensorflow/compiler/')
E2E=$(echo "${FILES?}" | grep '^\./integrations/tensorflow/e2e/')
```

Reply:
I wrote it that way because I couldn't seem to get `^[\./]?integrations` to work, but it looks like if I write `^\.?/?integrations` it will still work.

Reply:
(`find` and `git-diff` have different styles for the paths they output).

Reply:
Ah, `^(\./)?integrations` is what I wanted.